internal/events/release_succeeded.go

```diff
@@ -0,0 +1,35 @@
+package events
+
+import (
+	"encoding/json"
+
+	"github.com/lunarway/release-manager/internal/broker"
+)
+
+type ReleaseSucceeded struct {
+	Name          string `json:"name,omitempty"`
+	Namespace     string `json:"namespace,omitempty"`
+	ResourceType  string `json:"resourceType,omitempty"`
+	AvailablePods int32  `json:"availablePods,omitempty"`
+	DesiredPods   int32  `json:"desiredPods,omitempty"`
+	ArtifactID    string `json:"artifactID,omitempty"`
+	AuthorEmail   string `json:"authorEmail,omitempty"`
+	Environment   string `json:"environment,omitempty"`
+}
+
+// TODO: Make generic when go1.18 is adopted
+// Marshal implements broker.Publishable
+func (re ReleaseSucceeded) Marshal() ([]byte, error) {
+	return json.Marshal(re)
+}
+
+// Unmarshal implements broker.Publishable
+func (re ReleaseSucceeded) Unmarshal(data []byte) error {
```

Comment:
don't you want pointer-receiver here?

Reply:
This is the same with the other events. I think going for a generics implementation as noted elsewhere might be a way forward on this.